src/applications/bmqtool/m_bmqtool_application.h

```diff
@@ -167,6 +167,19 @@ class Application : public bmqa::SessionEventHandler {
     // message was seen during the current
     // grace period.
 
+    int d_numExpectedAcks;
+    // Auto-produce mode only. The total number of messages
+    // the tool will send. After posting is finished
+    // the tool will be waiting for this number of ACK
+    // messages, after which the shutdown semaphore will
+    // be posted.
+
+    int d_numPostedAcknowledged;
```

Comment:
Good catch in general!
Changing to bsl::uint64_t though, because:
1. I don't plan to store negative values there, so unsigned type is choosen;
2. ::uint64_t did not seem to be supported by C++03
3. bsls::Types::UInt64 is a legacy type

Reply:
I think `bsl::uint64_t` is a good choice, if you don't want to add a special meaning for negative values here.

Note that for `bmqbrkr` and related libs we preferred to use `bsls::Types::...` at least while we support AIX/Solaris, using `bsl::uint64_t` in `bmqtool` should be fine though